File: front/notas/src/app/dto/MateriaDTO.ts
import { usuarioDTO } from './usuarioDTO';
import { CursosDTO } from './CursosDTO';

export class MateriaDTO {

    public idMateria: number;
    public nombre: string;
    public abreviatura: string;
    public profesor: usuarioDTO;
    public idCurso: CursosDTO;
    public responsable:string;
}

File: back/notas/src/main/java/com/notas/dto/MateriaDTO.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.notas.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 *
 * @author UTP
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MateriaDTO {
    
    private Integer idMateria;
    private String nombre;
    private String abreviatura;
    private UsrUsuarioDTO profesor;
    private CursoDTO idCurso;
    private String responsable; 
}
